File: MiniDOMWebCtrls/DMD/CQSPD/CCQSPDImportExportHandler.cs

namespace minidom.Forms
{
    public class CCQSPDImportExportHandler : CBaseModuleHandler
    {
        public CCQSPDImportExportHandler() : base(ModuleSupportFlags.SCreate | ModuleSupportFlags.SDelete | ModuleSupportFlags.SEdit | ModuleSupportFlags.SAnnotations | ModuleSupportFlags.SExport)
        {
        }

        public override Databases.DBObjectCursorBase CreateCursor()
        {
            return new Finanziaria.CImportExportCursor();
        }

        public override object GetInternalItemById(int id)
        {
            return Finanziaria.ImportExport.GetItemById(id);
        }

        public string Esporta(object renderer)
        {
            Finanziaria.CImportExport item = (Finanziaria.CImportExport)DMD.XML.Utils.Serializer.Deserialize(this.n2str(renderer, "item"));
            item.Esporta();
            return DMD.XML.Utils.Serializer.Serialize(item);
        }

        public string ConfermaEsportazione(object renderer)
        {
            Finanziaria.CImportExport item = (Finanziaria.CImportExport)DMD.XML.Utils.Serializer.Deserialize(this.n2str(renderer, "item"));
            item.ConfermaEsportazione(item.MessaggioConferma);
            return DMD.XML.Utils.Serializer.Serialize(item);
        }

        public string Importa(object renderer)
        {
            Finanziaria.CImportExport item = (Finanziaria.CImportExport)DMD.XML.Utils.Serializer.Deserialize(this.n2str(renderer, "item"));
            item.Importa();
            return DMD.XML.Utils.Serializer.Serialize(item);
        }

        public string PrendiInCarico(object renderer)
        {
            Finanziaria.CImportExport item = (Finanziaria.CImportExport)DMD.XML.Utils.Serializer.Deserialize(this.n2str(renderer, "item"));
            item.PrendiInCarico();
            return DMD.XML.Utils.Serializer.Serialize(item);
        }

        public string Sincronizza(object renderer)
        {
            Finanziaria.CImportExport item = (Finanziaria.CImportExport)DMD.XML.Utils.Serializer.Deserialize(this.n2str(renderer, "item"));
            CCollection oggetti = (CCollection)DMD.XML.Utils.Serializer.Deserialize(this.n2str(renderer, "oggetti"));
            item.Sincronizza((CKeyCollection)oggetti);
            return DMD.XML.Utils.Serializer.Serialize(item);
        }

        public string Sollecita(object renderer)
        {
            Finanziaria.CImportExport item = (Finanziaria.CImportExport)DMD.XML.Utils.Serializer.Deserialize(this.n2str(renderer, "item"));
            item.Sollecita();
            return DMD.XML.Utils.Serializer.Serialize(item);
        }
    }
}

File: MiniDOMWebCtrls/DMD/CQSPD/CCQSPDImportExportHandler.vb
Imports minidom
Imports minidom.Forms
Imports minidom.Databases
Imports minidom.WebSite

Imports minidom.Finanziaria
Imports minidom.Sistema
Imports minidom.Anagrafica
Imports minidom.Forms.Utils
Imports minidom.CustomerCalls

Namespace Forms


    Public Class CCQSPDImportExportHandler
        Inherits CBaseModuleHandler

        Public Sub New()
            MyBase.New(ModuleSupportFlags.SCreate Or ModuleSupportFlags.SDelete Or ModuleSupportFlags.SEdit Or ModuleSupportFlags.SAnnotations Or ModuleSupportFlags.SExport)
        End Sub

        Public Overrides Function CreateCursor() As DBObjectCursorBase
            Return New CImportExportCursor()
        End Function

        Public Overrides Function GetInternalItemById(id As Integer) As Object
            Return Finanziaria.ImportExport.GetItemById(id)
        End Function


        Public Function Esporta(ByVal renderer As Object) As String
            Dim item As CImportExport = XML.Utils.Serializer.Deserialize(RPC.n2str(GetParameter(renderer, "item", "")))
            item.Esporta()
            Return XML.Utils.Serializer.Serialize(item)
        End Function

        Public Function ConfermaEsportazione(ByVal renderer As Object) As String
            Dim item As CImportExport = XML.Utils.Serializer.Deserialize(RPC.n2str(GetParameter(renderer, "item", "")))
            item.ConfermaEsportazione(item.MessaggioConferma)
            Return XML.Utils.Serializer.Serialize(item)
        End Function

        Public Function Importa(ByVal renderer As Object) As String
            Dim item As CImportExport = XML.Utils.Serializer.Deserialize(RPC.n2str(GetParameter(renderer, "item", "")))
            item.Importa()
            Return XML.Utils.Serializer.Serialize(item)
        End Function

        Public Function PrendiInCarico(ByVal renderer As Object) As String
            Dim item As CImportExport = XML.Utils.Serializer.Deserialize(RPC.n2str(GetParameter(renderer, "item", "")))
            item.PrendiInCarico()
            Return XML.Utils.Serializer.Serialize(item)
        End Function

        Public Function Sincronizza(ByVal renderer As Object) As String
            Dim item As CImportExport = XML.Utils.Serializer.Deserialize(RPC.n2str(GetParameter(renderer, "item", "")))
            Dim oggetti As CCollection = XML.Utils.Serializer.Deserialize(RPC.n2str(GetParameter(renderer, "oggetti", "")))
            item.Sincronizza(oggetti)
            Return XML.Utils.Serializer.Serialize(item)
        End Function

        Public Function Sollecita(ByVal renderer As Object) As String
            Dim item As CImportExport = XML.Utils.Serializer.Deserialize(RPC.n2str(GetParameter(renderer, "item", "")))
            item.Sollecita()
            Return XML.Utils.Serializer.Serialize(item)
        End Function

    End Class

End Namespace